src/test/java/com/squareup/javapoet/MethodSpecTest.java

```diff
@@ -83,74 +109,95 @@ private TypeElement getElement(Class<?> clazz) {
   @interface Nullable {
   }
 
-  @SuppressWarnings("unused") // Used via mirror API.
   abstract static class Everything {
     @Deprecated protected abstract <T extends Runnable & Closeable> Runnable everything(
         @Nullable String thing, List<? extends T> things) throws IOException, SecurityException;
   }
 
-  @SuppressWarnings("unused") // Used via mirror API.
   abstract static class HasAnnotation {
     @Override public abstract String toString();
   }
 
+  interface ExtendsOthers extends Callable<Integer>, Comparable<Long> {
+    // empty on purpose
+  }
+
   @Test public void overrideEverything() {
     TypeElement classElement = getElement(Everything.class);
+    DeclaredType classType = (DeclaredType) classElement.asType();
     ExecutableElement methodElement = getOnlyElement(methodsIn(classElement.getEnclosedElements()));
 
-    MethodSpec method = MethodSpec.overriding(methodElement).build();
+    MethodSpec method = MethodSpec.overriding(methodElement, classType, elements, types).build();
     assertThat(method.toString()).isEqualTo(""
         + "@java.lang.Override\n"
+        + "@java.lang.Deprecated\n"
         + "protected <T extends java.lang.Runnable & java.io.Closeable> "
-        + "java.lang.Runnable everything(java.lang.String arg0, java.util.List<? extends T> arg1) "
+        + "java.lang.Runnable everything("
+        + "@com.squareup.javapoet.MethodSpecTest.Nullable java.lang.String arg0, "
+        + "java.util.List<? extends T> arg1) "
         + "throws java.io.IOException, java.lang.SecurityException {\n"
         + "}\n");
-    // TODO see TODOs in MethodSpec.override
```

Comment:
?

First, all `TODO`s from `MethodSpec` resolved, next, all `TODO`s in `MethodSpecTest` ... vanished.